url47ebb27197_asirbu_CorrectiveEvolution_tgz-pJ8-eu_fbk_soa_evolution_engine_impl_test_ProblemToSMVFuncTestJ8/src/eu/fbk/soa/evolution/engine/impl/STSToProcessModel.java

```diff
@@ -0,0 +1,500 @@
+		simplifiedSTS.replaceState(state, replacement);
+		simplifiedSTS.removeState(state);
+			
+		Set<Transition> mappedTransitions = new HashSet<Transition>(
+				trans2node.keySet());
+		for (Transition trans : mappedTransitions) {
+			Transition newTrans = trans.replaceState(state, replacement);
+			if (!newTrans.equals(trans)) {
+				trans2node.put(newTrans, trans2node.get(trans));
+				trans2node.remove(trans);
+			}
+		}
+
+		Set<Transition> condTransitions = new HashSet<Transition>(
+				this.trans2cond.keySet());
+		for (Transition trans : condTransitions) {
+			Transition newTrans = trans.replaceState(state, replacement);
+			if (!newTrans.equals(trans)) {
+				trans2cond.put(newTrans, trans2cond.get(trans));
+				trans2cond.remove(trans);
+			}
+		}	
+	}
+	
+	
+	private boolean addAsActivityNode(Transition trans) {
+		Activity act = this.getCorrespondingActivity(trans);
+		if (act != null) {
+			ActivityNode actNode = new ActivityNode(act);
+			newPM.addVertex(actNode);
+			trans2node.put(trans, actNode);
+			return true;
+		}
+		return false;
+	}
+	
+	private boolean addAsXorSplit(Transition trans) {	
+		StateFormula cond1 = this.getCorrespondingCondition(trans);
+		State source = trans.getSource();
+		
+		if (!cond1.isEmpty()) {
+			boolean thereIsAnotherBranch = false;
+			
+			for (Transition t2 : this.simplifiedSTS.getTransitionsFromState(trans.getSource())) {
+				if (!t2.getCondition().equals(trans.getCondition()) || !t2.getTarget().equals(trans.getTarget())) {
+					thereIsAnotherBranch = true;
+				}
+			}
+			if (!thereIsAnotherBranch) {
+				return false;
+			}
+			
+			trans2cond.put(trans, cond1);
+			ProcessNode node = null;
+			for (Transition t2 : trans2cond.keySet()) {
+				if (!trans2node.containsKey(t2)) {
+					continue;
+				}
+				StateFormula cond2 = trans2cond.get(t2);				
+				if (t2.getSource().equals(source) && !cond1.equals(cond2)) {
+					node = trans2node.get(t2);
+					break;
+				}
+			}
+			if (node == null) {
+				node = new XorSplit();
+				this.newPM.addVertex(node);
+			}
+			trans2node.put(trans, node);
+			return true;
+		}
+		return false;
+	}
+	
+	
+	private void addEdges() {		
+		for (Transition trans : simplifiedSTS.getTransitions()) {
+			if (trans.getActionName() == "" || trans.getActionName().equals(defaultName)) {
+				continue;
+			}
+			State toState = trans.getTarget();
+			Set<Transition> connections = new HashSet<Transition>();
+			for (Transition outT : simplifiedSTS.getTransitionsFromState(toState)) {
+				connections.addAll(collectNextTransitions(outT));
+			}
+			for (Transition conn : connections) {
+				logger.trace("Adding edge for \n   " + trans.toString() + 
+						"\n   and " + conn.toString());
+				connectProcessNodes(trans, conn);
+			}
+		}
+	}
+	
+	private Set<Transition> collectNextTransitions(Transition trans) {
+		Set<Transition> nextTrans = new HashSet<Transition>();
+		
+		if (trans.getActionName() != "" && !trans.getActionName().equals(defaultName)) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.